middleware/markdown/process.go

```diff
@@ -1,216 +1,45 @@
-func (md Markdown) processTemplate(c *Config, requestPath string, tmpl []byte, metadata Metadata, ctx middleware.Context) ([]byte, error) {
-	// if template is not specified,
-	// use the default template
-	if tmpl == nil {
-		tmpl = defaultTemplate(c, metadata, requestPath)
-	}
-
-	// process the template
-	b := new(bytes.Buffer)
-	t, err := template.New("").Parse(string(tmpl))
-	if err != nil {
-		return nil, err
-	}
-	mdData := Data{
-		Context:  ctx,
-		Doc:      metadata.Variables,
-		DocFlags: metadata.Flags,
-		Links:    c.Links,
-	}
-
-	c.RLock()
-	err = t.Execute(b, mdData)
-	c.RUnlock()
-
-	if err != nil {
-		return nil, err
-	}
-
-	// generate static page
-	if err = md.generatePage(c, requestPath, b.Bytes()); err != nil {
-		// if static page generation fails, nothing fatal, only log the error.
-		// TODO: Report (return) this non-fatal error, but don't log it here?
-		log.Println("[ERROR] markdown: Render:", err)
-	}
-
-	return b.Bytes(), nil
-
-}
-
-// generatePage generates a static html page from the markdown in content if c.StaticDir
-// is a non-empty value, meaning that the user enabled static site generation.
-func (md Markdown) generatePage(c *Config, requestPath string, content []byte) error {
-	// Only generate the page if static site generation is enabled
-	if c.StaticDir != "" {
-		// if static directory is not existing, create it
-		if _, err := os.Stat(c.StaticDir); err != nil {
-			err := os.MkdirAll(c.StaticDir, os.FileMode(0755))
-			if err != nil {
-				return err
-			}
-		}
-
-		// the URL will always use "/" as a path separator,
-		// convert that to a native path to support OS that
-		// use different path separators
-		filePath := filepath.Join(c.StaticDir, filepath.FromSlash(requestPath))
-
-		// If it is index file, use the directory instead
-		if md.IsIndexFile(filepath.Base(requestPath)) {
-			filePath, _ = filepath.Split(filePath)
-		}
-
-		// Create the directory in case it is not existing
-		if err := os.MkdirAll(filePath, os.FileMode(0744)); err != nil {
-			return err
-		}
-
-		// generate index.html file in the directory
-		filePath = filepath.Join(filePath, "index.html")
-		err := ioutil.WriteFile(filePath, content, os.FileMode(0664))
-		if err != nil {
-			return err
-		}
-
-		c.Lock()
-		c.StaticFiles[requestPath] = filepath.ToSlash(filePath)
-		c.Unlock()
-	}
+	mdata.Variables["title"] = title
 
-	return nil
-}
-
-// defaultTemplate constructs a default template.
-func defaultTemplate(c *Config, metadata Metadata, requestPath string) []byte {
-	var scripts, styles bytes.Buffer
-	for _, style := range c.Styles {
-		styles.WriteString(strings.Replace(cssTemplate, "{{url}}", style, 1))
-		styles.WriteString("\r\n")
-	}
-	for _, script := range c.Scripts {
-		scripts.WriteString(strings.Replace(jsTemplate, "{{url}}", script, 1))
-		scripts.WriteString("\r\n")
+	if len(dirents) > 0 {
+		mdata.Flags["dirents"] = true
+		mdata.Dirents = dirents
+	} else {
+		mdata.Flags["dirents"] = false
```

Comment:
The flag is for convenience in templates.  Do templates evaluate zero length arrays as false?  IE:

```
{{ if .Dirents}}
<ul>
{{ range .Dirents}}
  <li>{{.Name}}</li>
{{ end }}
{{ fi }}
```
Or how that looks in actual template.  :)